### ISSUE TITLE
add compat to dependencies

### ISSUE THREAD

**Issue by kafisatz** [xiaodaigh/FastGroupBy.jl#10]
When using your package, I get an error saying that you should add Compat to your dependencies:
```
julia> using FastGroupBy
[ Info: Precompiling FastGroupBy [e6aeac8e-85c6-520e-8612-effc59bc0f33]
┌ Warning: Package FastGroupBy does not have Compat in its dependencies:
│ - If you have FastGroupBy checked out for development and have
│   added Compat as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with FastGroupBy
└ Loading Compat into FastGroupBy from project dependency, future warnings for FastGroupBy are suppressed.
```

**Comment by xiaodaigh** [comment]
can you try `]dev FastGroupBy`. I don't' get the same error in the latest dev version on Julia 1.1.1.

**Comment by kafisatz** [comment]
I am on 1.0.4 for various reasons. And will stay on that version for a while (I think)

**Comment by xiaodaigh** [comment]
> I am on 1.0.4 for various reasons

Is `]dev FastGroupBy` an option for you? I might look at get a new release soon

**Comment by kafisatz** [comment]
Thanks that does actually work for me.

However, I noted that there is some additional functionality which your package does currently not offered (compared to DataFrames.aggregate and DataFrames.groupby) which are:
1) Considering your last example in the Readme: if I have several :grps, say I want to aggregate by `[:grps1, :grps2, :grps3]` that does not seem possible in this package.
2) if :grps is a custom type which I have defined (for which I have defined isless and thus sort). FastBy does not seem to work. I realize that I may miss some (or all) of the speedup of this package in this case.

Thank you for the support though.
